src/components/Dropdown/PrioritySelector/PrioritySelector.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import Close from '@material-design-icons/svg/round/close.svg';
+import KeyboardDoubleArrowUp from '@material-design-icons/svg/round/keyboard_double_arrow_up.svg';
+import KeyboardArrowUp from '@material-design-icons/svg/round/keyboard_arrow_up.svg';
+import KeyboardArrowDown from '@material-design-icons/svg/round/keyboard_arrow_down.svg';
+import { bem } from '../../../utils/bem';
+import { Dropdown } from '../Dropdown';
+import { ListItem } from '../../List';
+import { Text } from '../../Text';
+import styles from './PrioritySelector.scss';
+import { IconButton } from '../../Buttons';
+
+const iconMap = {
+    mandatory: KeyboardDoubleArrowUp,
+    important: KeyboardArrowUp,
+    optional: KeyboardArrowDown,
+    exclude: Close,
+};
+
+export type Priority = 'mandatory' | 'important' | 'optional' | 'exclude';
+
+export type PriorityItem = {
+    priority: Priority;
+    label: string;
+    value?: string;
+};
+
+const defaultList: PriorityItem[] = [
+    { priority: 'mandatory', label: 'Mandatory' },
+    { priority: 'important', label: 'Important' },
+    { priority: 'optional', label: 'Optional' },
+    { priority: 'exclude', label: 'Exclude' },
+];
+
+export interface Props<>extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
+    /** Currently selected priority item that indicates the importance of the component. */
+    selectedItem: PriorityItem;
+    /** Array of availible priority items. */
+    list: PriorityItem[];
+    /** Callback function triggered when a new priority is selected. */
+    onChange: (newPriorityItem: PriorityItem) => void;
+    /** Boolean indicating whether the whole badge should be disabled. */
+    isDisabled?: boolean;
+    /** Priority button label name for ARIA labelling */
+    buttonLabel?: string;
+}
+
+const { block, elem } = bem('PrioritySelector', styles);
+export function PrioritySelector({
+    selectedItem,
+    isDisabled = false,
+    onChange,
+    buttonLabel,
+    list = defaultList,
+    ...rest
+}: Props) {
+    const badgeRef = React.useRef<HTMLDivElement | null>(null);
```

Comment:
`badgeRef` should come from props since, for example, we may need to align the dropdown width with the entire `SelectedPriorityBadge`, and we don’t need any wrapper in this component. It should also be optional, so if specific width alignment isn’t needed, it will use the default behavior specified in the `Dropdown` component.

Reply:
omg. This is what I was talking about this morning! Oops! ;)